dashboard/app/helpers/javalab_files_helper.rb

```diff
@@ -55,9 +51,10 @@ def self.get_project_files(channel_id, level_id)
   #   "validation": <all validation code for a project, in json format>
   # }
   # If the level doesn't have validation and/or a maze, those fields will not be present.
-  def self.get_project_files_with_override_sources(sources, level_id)
```

Comment:
Looking at this class again, I wonder if we can streamline and combine these different functions by conditionally creating the files object just based on input? Something like
```
get_project_files(channel_id, level_id, sources, validation)
  all_files = get_level_files(level_id)
  if sources
     // add sources
  else
     // get sources from bucket
  if validation
     // add validation
  if channel_id
    // add assets
```
I'm good to ship this as is, but think this might be helpful especially as we consider moving to using the override sources path logic as the default behavior

Reply:
I think I'm going to leave this as-is because doing that refactor would require a lot of testing. I can file something on the backlog.

Reply:
https://codedotorg.atlassian.net/browse/SL-315